pkg/log/zap/zap_test.go

```diff
@@ -296,3 +299,170 @@ var _ = Describe("Zap logger setup", func() {
 		})
 	})
 })
+
+var _ = Describe("Zap log level flag options setup", func() {
+	var (
+		fromFlags Options
+		fs        flag.FlagSet
+		info      = "info text"
+		debug1    = "debug 1 text"
+		debug2    = "debug 2 text"
+		debug3    = "debug 3 text"
+	)
+
+	BeforeEach(func() {
+		fromFlags = Options{}
+		fs = *flag.NewFlagSet(os.Args[0], flag.ExitOnError) //flags are now reset
+	})
+
+	Context("with  zap-log-level options provided", func() {
+		It("Should set info/debug zap-log-level flags.", func() {
```

Comment:
The tests cases here are not clear. What should occur when `set info/debug zap-log-level flags`?

See how it is in the master: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/log/zap/zap_test.go

```
var _ = Describe("Zap logger setup", func() {
	Context("with the default output", func() {
		It("shouldn't fail when setting up production", func() {
....

	Context("when logging kubernetes objects", func() {
		var logOut *bytes.Buffer
		var logger logr.Logger

		defineTests := func() {
			It("should log a standard namespaced Kubernetes object name and namespace", func() {
				pod := &kapi.Pod{}
				pod.Name = "some-pod"

```

We should have similar tests and the descriptions should make easier understand what is testing and what should be the expected result. 

Also, should we not keep the same tests and just adapt them when required and add the new use cases that should be covered instead of remove all?